File: python/caesar cipher/decrypt.py
def decrypt_caesar(ciphertext):
    for key in range(1, 27):
        plaintext = ''
        for c in ciphertext:
            if c.isalpha():
                base = ord('A') if c.isupper() else ord('a')
                decrypted = chr(base + (ord(c) - base - key + 26) % 26)
                plaintext += decrypted
            else:
                plaintext += c  # Non-alphabetic characters remain unchanged
        print(f"Key {key} output: {plaintext}")

def main():
    ciphertext = input("Enter the ciphertext: ")
    decrypt_caesar(ciphertext)

if __name__ == "__main__":
    main()

File: c++/caesar cipher/decrypt.cpp
#include <iostream>
#include <string>

void decryptCaesar(const std::string& ciphertext, int shift) {
    for (int key = 1; key <= 26; ++key) {
        std::string plaintext;
        for (char c : ciphertext) {
            if (std::isalpha(c)) {
                char base = std::isupper(c) ? 'A' : 'a';
                char decrypted = base + (c - base - key + 26) % 26;
                plaintext += decrypted;
            } else {
                plaintext += c; // Non-alphabetic characters remain unchanged
            }
        }
        std::cout << "Key " << key << " output: " << plaintext << '\n';
    }
}

int main() {
    std::string ciphertext;
    std::cout << "Enter the ciphertext: ";
    std::cin >> ciphertext;

    decryptCaesar(ciphertext, 0); // Brute-force all keys
    return 0;
}
